public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -127,13 +126,13 @@ const Policy: FC<PolicyComponentProps> = ({
   const isEditable = canEditRoutes;
   const isDeletable = canDeleteRoutes && !isDefaultPolicy;
 
-  const matchingAlertGroups = useMemo(
-    () => findMatchingAlertGroups(routeTree, currentRoute, alertGroups),
-    [alertGroups, currentRoute, routeTree]
-  );
+  // const matchingAlertGroups = useMemo(
```

Comment:
Can we add a comment or link https://github.com/grafana/grafana/pull/61952#issuecomment-1437010087 so someone that hasn't seen this PR is aware of why it's commented in the code? 
Another thought, would it be worth it to execute this only when the tree is small/there are not a lot of firing instances? Not sure how easy that would be to calculate though.

Reply:
I've merged with main and hooked it up to the new feature toggles so we can ship the feature but behind a flag 👍